fem/tmop.cpp

```diff
@@ -4376,143 +4460,167 @@ void TMOP_Integrator::ComputeMinJac(const Vector &x,
    dx = detv_avg_min / dxscale;
 }
 
-void TMOP_Integrator::
-UpdateAfterMeshPositionChange(const Vector &x_new,
-                              const FiniteElementSpace &x_fes)
+void TMOP_Integrator::RemapSurfaceFittingLevelSetAtNodes(const Vector &new_x,
+                                                         int new_x_ordering)
 {
-   if (discr_tc) { PA.Jtr_needs_update = true; }
-
-   if (PA.enabled) { UpdateCoefficientsPA(x_new); }
-
-   Ordering::Type ordering = x_fes.GetOrdering();
-
-   // Update the finite difference delta if FD are used.
-   if (fdflag) { ComputeFDh(x_new, x_fes); }
-
-   // Update the target constructor if it's a discrete one.
-   if (discr_tc)
+   if (!surf_fit_gf) { return; }
+   if (surf_fit_gf_bg &&
+       dynamic_cast<InterpolatorFP *>(surf_fit_eval) &&
+       dynamic_cast<InterpolatorFP *>(surf_fit_eval_bg_grad) &&
+       dynamic_cast<InterpolatorFP *>(surf_fit_eval_bg_hess))
```

Comment:
What if one doesn't provide the `grad` and `hess` ones? They seemed to be optional.